### PR TITLE
Update and fix `connectivity` docs

### PR DESCRIPTION
**Pull request by user27182**
### Overview
Follow up from #4824. Improve docs and include minor bug fixes.

### Details
- Fix `compute-volume` example to use the correct `connectivity` `scalar_range` (percentages taken from `threshold_percent` were previously incorrectly used)
- Fix list formatting for `versionadded` directive and `variable_input` param (lists were previously formatted as a paragraph)
- Remove single float use case for `scalar_range` (must be a sequence with two values) and add relevant test cases
- Update `See Also` section with relevant methods (methods are now also hyperlinked)
- Move long examples to `examples/01-filter/connectivity.py`
- Add new short docstring examples
- Other minor documentation fixes to improve readability / clarity
- Fix bug with `'specified'` mode where specifying an unused region id would raise an unexpected `KeyError`. The filter now correctly returns an empty mesh.
- Fix bug with `PolyData.clean` where it would would raise an error when cleaning a mesh with points but no cells

### CODE REVIEW

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thank you for suggesting the correction, I do have one suggestion. The docstring Example section of connectivity seems a bit long in my mind. Could you move that content to `examples/01-filter/connectivity.py`? For us, the docstring is an introduction to using the method by default in a simple way, and we would like to see a separate example page for more detailed usage.

**Comment by user27182** [comment]
> Could you move that content to `examples/01-filter/connectivity.py`?

Ok. I agree this is a better place for these. The examples have been moved with some edits / changes, and new, simpler, docstring examples added. See the updated PR Overview for details.

File: examples/01-filter/connectivity.py [hunk 1]
```diff
@@ -4,39 +4,167 @@
...
+mesh = examples.download_foot_bones()
+
+###############################################################################
+# Extract all regions.
+conn = mesh.connectivity('all')
+
+###############################################################################
+# Plot the labelled regions.
+
+# Format scalar bar text for integer values.
```
Comment (tkoyama010) [inline examples/01-filter/connectivity.py:101]:
```suggestion
# Plot the labeled regions.

# Format scalar bar text for integer values.
```

File: examples/01-filter/connectivity.py [hunk 2]
```diff
@@ -4,39 +4,167 @@
...
+###############################################################################
+# Extract a region of interest using the
+# ``'point_seed'`` ``extraction_mode`` of the :func:`~pyvista.DataSetFilters.connectivity`
+# filter. Let's extract the steepest peak using a seed point where the
+# curvature is maximized.
+
+# Get seed point
+peak_point_id = np.argmax(mesh['Curvature'])
+
+# Define scalar range of region to extract
```
Comment (tkoyama010) [inline examples/01-filter/connectivity.py:149]:
```suggestion
# Define a scalar range of the region to extract
```

File: tests/core/test_polydata.py [hunk 3]
```diff
@@ -5,6 +5,7 @@
 import numpy as np
 import pytest
 
+import pyvista
```
Comment (tkoyama010) [inline tests/core/test_polydata.py:8]:
```suggestion
```

File: tests/core/test_polydata.py [hunk 4]
```diff
@@ -707,6 +708,12 @@ def test_clean(sphere):
     cleaned = mesh.clean(point_merging=False)
     assert cleaned.n_points == mesh.n_points
 
+    # test with points but no cells
+    mesh = pyvista.PolyData()
```
Comment (tkoyama010) [inline tests/core/test_polydata.py:712]:
```suggestion
    mesh = pv.PolyData()
```

File: pyvista/core/filters/data_set.py [hunk 5]
```diff
@@ -2498,132 +2502,60 @@ def connectivity(
...
 
-        >>> p = pv.Plotter()
-        >>> _ = p.add_mesh(conn)
-        >>> _ = p.add_mesh(mesh, style='wireframe')
-        >>> p.show(cpos=cpos)
+        >>> conn = mesh.connectivity('closest', (0, 0, 0))
+        >>> conn.plot(color='blue', show_edges=True)
 
-        Extract regions using seed points.
+        Extract a region using a cell id ``100`` as a seed.
```
Comment (tkoyama010) [inline pyvista/core/filters/data_set.py:2540]:
```suggestion
        Extract a region using a cell ID ``100`` as a seed.
```

File: pyvista/core/filters/poly_data.py [hunk 6]
```diff
@@ -657,6 +657,9 @@ def intersection(self, mesh, split_first=True, split_second=True, progress_bar=F
     def curvature(self, curv_type='mean', progress_bar=False):
         """Return the pointwise curvature of a mesh.
 
+        See :ref:`_connectivity_seed_ref` for more examples using this
+        filter.
```
Comment (tkoyama010) [inline pyvista/core/filters/poly_data.py:661]:
```suggestion
        See :ref:`_connectivity_example` for more examples using this
        filter.
```

**Review by tkoyama010 — APPROVED** [review]
LGTM. Thanks!

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**